lib/Sema/CSBindings.cpp

```diff
@@ -1722,42 +1722,83 @@ void BindingSet::dump(llvm::raw_ostream &out, unsigned indent) const {
 
   auto numDefaultable = getNumViableDefaultableBindings();
   if (numDefaultable > 0)
-    out << "#defaultable_bindings: " << numDefaultable << " ";
+    out << "[#defaultable_bindings: " << numDefaultable << "] ";
+
+  struct PrintableBinding {
+  private:
+    enum class BindingKind { Exact, Subtypes, Supertypes, Literal };
+    BindingKind Kind;
+    Type BindingType;
+    PrintableBinding(BindingKind kind, Type bindingType) {
+      Kind = kind;
+      BindingType = bindingType;
+    };
```

Comment:
Very nit, but in c++ there is `Initializer List`
```suggestion
    PrintableBinding(BindingKind kind, Type bindingType): BindingType(bindingType), Kind(kind){}
```